zetta_utils/common/semaphores.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import os
+from typing import List, Literal, get_args
+
+import attrs
+from posix_ipc import (  # pylint: disable=no-name-in-module
+    O_CREX,
+    ExistentialError,
+    Semaphore,
+)
+
+from zetta_utils import log
+
+logger = log.get_logger("zetta_utils")
+SemaphoreType = Literal["read", "write", "cuda", "cpu"]
+
+DEFAULT_SEMA_COUNT = 1
+
+
+def name_to_posix_name(name: SemaphoreType, pid: int) -> str:  # pragma: no cover
+    return f"zetta_utils_{pid}_{name}_semaphore"
+
+
+class configure_semaphores:  # pylint: disable=invalid-name
```

Comment:
Same here, I think using `contextlib`'s decorator to make it a function will get rid of the invalid-name